src/renderer/utils/TestSequenceProjectHandler.ts

```diff
@@ -0,0 +1,278 @@
+// Exposed function in this file should alway follow the following pattern:
+// Take the project as an Input.
+//   1. Always valide the state of the proejct
+//   2. Do the operation
+//   3. Update the test sequencer
+//   4. Return a Result<T, Error>
+// Note: All path in a project should be using '/' for cross-platform compatibility
+
+import { Err, Ok, Result } from "@/types/result";
+import { readJsonProject, stringifyProject } from "@/renderer/routes/test_sequencer_panel/utils/TestSetUtils";
+import { TestSequenceElement, TestSequencerProject } from "@/renderer/types/testSequencer";
+
+
+// Exposed API
+export type StateManager = {
+  elem: TestSequenceElement[];
+  setElems: (elems: TestSequenceElement[]) => void;
+  project: TestSequencerProject | null;
+  setProject: (project: TestSequencerProject | null) => void;
+  setUnsaved: (unsaved: boolean) => void;
+}
+
+export async function createProject(project: TestSequencerProject, stateManager: StateManager, throwInsteadOfResult: boolean=false): Promise<Result<null, Error>> {
+  // Create a new project from the element currently in the test sequencer
+  // - Will valide that each element is in the base folder
+  try {
+    const elems = stateManager.elem;
+    project = validatePath(project);
+    project = updateProjectElements(project, await createProjectElementsFromTestSequencerElements(elems, project.projectPath, true));
+    await saveToDisk(project);
+    await syncProject(project, stateManager);
+    return Ok(null);
+  } catch (e: unknown) {
+    if (throwInsteadOfResult) throw e;
+    return buildResultFromCatch(e); 
+  }
+}
+
+export async function saveProject(stateManager: StateManager, throwInsteadOfResult: boolean=false): Promise<Result<null, Error>> {
+  // Save the current project to disk
+  try {
+    let project = stateManager.project;
+    if (!project) {
```

Comment:
Agree, but I feel like I'm exposing a problem. Better validate the state and don't expect the caller to be smart. (Since it's in a separate file, we could infer that somewhere else could call this at some point)